### DISCUSSION TITLE
µGrants

### DISCUSSION THREAD

**Discussion by SergioBenitez** [rwf2/rwf2#8] [Ideas]
It is my hope that we can dramatically increase high-quality contributions to Rocket and associated projects through the foundation's sponsorship of µGrants: small (≤ $1k) grants for concrete work. A proposal for their structure can be found at [docs/micro-grants.md](https://github.com/rwf2/rwf2.org/blob/master/docs/micro-grants.md).

Please share your thoughts and suggestions. What feels right? What feels wrong?

**Comment by Qwuke** [comment]
I think the document is well thought out, especially how explicit the steps and requirements are for individuals wanting to get started right away.

Early termination could leave a contributor feeling burned if it were to ever catch them off-guard - does the "definite time-frame" requirement mentioned at the start involve a deadline or "reasonable time-frame" guidance as part of the grant proposal/application process? I think some upfront communication or guidance about that could help prevent ruffled feathers, but I think specific dates as deadlines would only demonstrate the difficulty of software estimation. 

When talking about project time-frames, you mention they should be "unconstrained by external agenda": that phrase could definitely be made clearer. I'm not certain but I believe you're saying that timelines for grants should be determined by the work required and not by a potential grantee's time schedule or other factors - which is a good guideline.

**Reply by SergioBenitez** [reply]
Thank you for the feedback! This is really helpful.

> Early termination could leave a contributor feeling burned if it were to ever catch them off-guard

Early termination is definitely not designed to be a threat! The idea with early termination is only to have a codified process by which granted proposals making no progress can be revived. By "no progress" I mean two things: 1) the implementor cannot be reached after several attempts, or 2) the implementor signals that they are unable to make any further progress.

> When talking about project time-frames, you mention they should be "unconstrained by external agenda": that phrase could definitely be made clearer.

You're absolutely right. I'll work on that now. But yes, I do mean what you're suggesting, but I also meant that the project shouldn't have goals that depend on external factors outside of our control. For example, a project that requires working across different crates owned by different organizations is unlikely to be a good candidate for a project as making progress would depend on those other organizations accepting our changes. Instead, those changes should be made to the other crates before a uGrant is awarded.

**Reply by SergioBenitez** [reply]
I've updated the document with consideration for your feedback. Thank you again!
